tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/Models/AzureSearchChatExtensionConfiguration.Serialization.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+namespace AzureOpenAI.Models;
+
+public partial class AzureSearchChatExtensionConfiguration :  IJsonModel<AzureSearchChatExtensionConfiguration>
```

Comment:
Note, some models like these are only in the Azure package.  But we assign instances of them at runtime to objects in the unbranded package and reference them as IJsonModel.  Think a bit more about this and the implications of it.

Reply:
One angle: what does the generator do?
Side question on that - how does the Tsp represent the differences so the generator can do a different thing?